app/workers/document_list_export_worker.rb

```diff
@@ -15,6 +15,8 @@ def perform(filter_options, user_id)
 
   def send_mail(csv_filename, user, filter)
     MailNotifications.document_list(csv_filename, user.email, filter.page_title).deliver_now
+  rescue Notifications::Client::BadRequestError
```

Comment:
I wonder if we could check if the error is the not in team error (regex match on the exception message maybe?) and if not send it to Sentry. Because if it was something else (incorrect API key for example) in production, it might be difficult to debug without these exeptions.

Reply:
I wonder if you could look at the [`SENTRY_CURRENT_ENV` ENV variable](https://github.com/alphagov/govuk_app_config#automatic-error-reporting) from within the worker, and only call `GovukError.notify` if it's not 'integration' or 'staging'? (I deliberately suggest that rather than "only if it's 'production'" in case the variable goes missing one day and we lose visibility on production errors).

Reply:
Yeah, that sounds like a good idea.

I also think that actually in production we should simply `raise` rather than using `GovukError.notify` directly so we retain Sidekiq's retry behaviour if it's just a transient issue.

Reply:
Just to clarify the intended logic here: do you mean if it's not the not-in-team error _and_ it's production, or do you mean if it's not the not-in-team error _or_ it's production?

i.e., we don't care about not-in-team errors in any environment; we care about all errors in production; but do we care about other bad-request errors in integration/staging?

Reply:
My thinking was:

- In production: we care about _all_ errors, including the not-in-team ones (although why they would appear in production... I'm not sure... only a misconfiguration I imagine, which would still be a useful thing to know about)
- In the other environments: we don't care about the not-in-team errors, but we do care about everything else

Reply:
I'd ideally remove this change from the first commit, making it just about warning the user. Then in your next commit you have the exception handling change.